demos/python_demos/object_detection_demo_ssd_async/object_detection_demo_ssd_async.py

```diff
@@ -15,241 +15,581 @@
+            self.callback_exceptions.append(e)
+        self.event.set()
+
+    def __call__(self, inputs, id, meta):
+        request = self.empty_requests.popleft()
+        inputs = self.unify_inputs(inputs)
+        inputs, preprocessing_meta = self.preprocess(inputs)
+        meta.update(preprocessing_meta)
+        request.async_infer(inputs=inputs)
+        request.set_completion_callback(py_callback=self.request_callback,
+                                        py_data=(request, id, meta))
+
+    def await_all(self):
+        for request in self.exec_net.requests:
+            request.wait()
+
+    def await_any(self):
+        self.event.wait()
+
+
+class Detector(Model):
+
+    class Detection:
+        def __init__(self, xmin, ymin, xmax, ymax, score, class_id):
+            self.xmin = xmin
+            self.xmax = xmax
+            self.ymin = ymin
+            self.ymax = ymax
+            self.score = score
+            self.class_id = class_id
+
+    def __init__(self, *args, labels_map=None, keep_aspect_ratio_resize=False, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.keep_aspect_ratio_resize = keep_aspect_ratio_resize
+        self.labels_map = labels_map
+
+        self.image_blob_name, self.image_info_blob_name = self._get_inputs(self.net)
+        self.n, self.c, self.h, self.w = self.net.input_info[self.image_blob_name].input_data.shape
+        assert self.n == 1, 'Only batch size == 1 is supported.'
+
+        self.output_parser = self._get_output_parser(self.net, self.image_blob_name)
+
+    def _get_inputs(self, net):
+        image_blob_name = None
+        image_info_blob_name = None
+        for blob_name, blob in net.input_info.items():
+            if len(blob.input_data.shape) == 4:
+                image_blob_name = blob_name
+            elif len(blob.input_data.shape) == 2:
+                image_info_blob_name = blob_name
+            else:
+                raise RuntimeError('Unsupported {}D input layer "{}". Only 2D and 4D input layers are supported'
+                                   .format(len(blob.shape), blob_name))
+        assert image_blob_name is not None
+        return image_blob_name, image_info_blob_name
+
+    def _get_output_parser(self, net, image_blob_name, bboxes='bboxes', labels='labels', scores='scores'):
+        if len(net.outputs) == 1:
+            output_blob = next(iter(net.outputs))
+            log.info('Use SingleOutputParser')
+            return SingleOutputParser(output_blob)
+        else:
+            try:
+                parser = MultipleOutputParser(net.outputs, bboxes, scores, labels)
+                log.info('Use MultipleOutputParser')
+            except ValueError:
+                h, w = net.input_info[image_blob_name].input_data.shape[2:]
+                parser = OTEParser([w, h], net.outputs)
+                log.info('Use OTEParser')
+            return parser
+        raise RuntimeError('Unsupported model outputs')
```

Comment:
```suggestion
```
This is never reached.

Reply:
Changed the logic a bit here.

Reply:
`else` up there seems to be redundant.

Reply:
Thanks. Moved some checks to `SingleOutputParser` itself, made parser selection more consistent.